hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/AbstractOzoneFileSystemTest.java

```diff
@@ -2230,7 +2236,17 @@ void testFileSystemWithObjectStoreLayout() throws IOException {
       OzoneConfiguration config = new OzoneConfiguration(fs.getConf());
       config.set(FS_DEFAULT_NAME_KEY, obsRootPath);
 
-      IllegalArgumentException e = assertThrows(IllegalArgumentException.class, () -> FileSystem.get(config));
+      IllegalArgumentException e =
+          assertThrows(IllegalArgumentException.class, () -> {
+            FileSystem fileSystem = null;
+            try {
+              fileSystem = FileSystem.get(config);
+            } finally {
+              if (fileSystem != null) {
+                fileSystem.close();
+              }
+            }
```

Comment:
Let's add a utility method (e.g. in `GenericTestUtils`) to close the object if expected exception is not thrown.

```java
  public static <T extends Throwable> T assertThrows(
      Class<T> expectedType,
      Callable<? extends AutoCloseable> func) {
    return Assertions.assertThrows(expectedType, () -> {
      final AutoCloseable closeable = func.call();
      try {
        if (closeable != null) {
          closeable.close();
        }
      } catch (Exception ignored) {
      }
    });
  }
```

This let's us keep existing assertion statements (except we need to qualify the call with `GenericTestUtils`:

```java
IllegalArgumentException e = GenericTestUtils.assertThrows(IllegalArgumentException.class,
    () -> FileSystem.get(config));
```

We can also reuse it e.g. for assertions about I/O:

https://github.com/apache/ozone/blob/d1e6900b957348ce50bc13cc5e30df3e1b7605fe/hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/AbstractOzoneFileSystemTest.java#L412-L414